src/note_position.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+use std::iter::Peekable;
+
+/// An iterator adaptor that wraps each element in an [`Position`](../enum.Position.html).
+///
+/// Iterator element type is `Position<I::Item>`.
+/// This iterator is *fused*.
```

Comment:
Yuck. FWIW, I think I misread the std docs, which say `impl<I> FusedIterator for Peekable<I> where I: FusedIterator`, which makes sense.

Should I change this to `Peekable<Fuse<I>>` to work around that problem for now?